### PR TITLE
Updating Traffic Generators

### PR DESCRIPTION
**Pull request by MMysore2**
Added documentation for `strided_generator.py` and `strided_generator_core.py.`

Updated clarity of documentation for `linear_generator.py`, `linear_generator_core.py`, `random_generator.py`, and `random_generator_core.py`.

Made `max_addr` exclusive instead of inclusive for strided and linear traffic generation in `strided_gen.cc` and `linear_gen.cc`.

### CODE REVIEW

**Comment by ivanaamit** [comment]
Hi @MMysore2,

None of the commits in this pull request contains a Change-ID, which we require for any changes made to gem5.  To automatically insert one, run the following:

```
f=.git/hooks/commit-msg
mkdir -p $f
curl -Lo $f https://gerrit-review.googlesource.com/tools/hooks/commit-msg
chmod +x $f
```

Then, amend the commit with `git commit --amend --no-edit`, and update your pull request. Thank you.

**Comment by MMysore2** [comment]
> Hi @MMysore2,
> 
> None of the commits in this pull request contains a Change-ID, which we require for any changes made to gem5. To automatically insert one, run the following:
> 
> ```
> f=.git/hooks/commit-msg
> mkdir -p $f
> curl -Lo $f https://gerrit-review.googlesource.com/tools/hooks/commit-msg
> chmod +x $f
> ```
> 
> Then, amend the commit with `git commit --amend --no-edit`, and update your pull request. Thank you.

Hi @ivanaamit!

I just made the changes, so let me know if there's anything else to add.

Thank you!

File: src/python/gem5/components/processors/linear_generator.py [hunk 1]
```diff
@@ -64,7 +64,7 @@ def __init__(
         generator cores that could replace the processing cores in a board.
 
         :param num_cores: The number of linear generator cores to create.
-        :param duration: The number of ticks for the generator to generate
+        :param duration: The duration of time for which the generator generates
```
Comment (BobbyRBruce) [inline src/python/gem5/components/processors/linear_generator.py:67]: Is there anything you can reference so somneone reading this knows the format of the string to pass? Reading this my initial feeling is "what's the unit here?" but the parameter is a string so I assume it's "<integer><SI unit>s"?

Maybe a bit nitpicky, but would be nice to be specific.

((not in the scope for this PR but I think having duration in ticks is much better. That or an int or float for the number of, say, nano seconds to simulate. I really think a string is a terrible type for duration...))

Comment (MMysore2) [reply src/python/gem5/components/processors/linear_generator.py:67]: Hi @BobbyRBruce, I just updated the documentation for the duration parameter.

I also added some documentation for the data_limit parameter.

These changes are under the latest commit.

Let me know if there's anything else you'd like to see!

(I also agree that having an int or float for duration might be more intuitive.)

Thank you!

**Review by BobbyRBruce — APPROVED** [review]

**Pull request merged**